lib/sdf-server/src/service/v2/index.rs

```diff
@@ -85,7 +89,51 @@ pub async fn get_change_set_index(
     let change_set = ChangeSet::get_by_id(&ctx, change_set_id).await?;
 
     let index = match frigg.get_index(workspace_pk, change_set_id).await? {
-        Some((index, _kv_revision)) => index,
+        Some((index, kv_revision)) => {
+            let mv_index: MvIndex = serde_json::from_value(index.data.to_owned())
+                .map_err(IndexError::DeserializingMvIndexData)?;
+
+            let mut implemented_kinds = HashSet::new();
+            for index_ref in mv_index.mv_list {
+                let kind = ReferenceKind::try_from(index_ref.kind.as_str())
+                    .map_err(IndexError::InvalidStringForReferenceKind)?;
+                if kind.is_revision_sensitive() {
+                    implemented_kinds.insert(kind);
+                }
+            }
```

Comment:
Probably worth adding a span around this to have some information on how costly this bit is over time.